src/umpire/strategy/AlignedAllocator.cpp

```diff
@@ -0,0 +1,71 @@
+//////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2016-20, Lawrence Livermore National Security, LLC and Umpire
+// project contributors. See the COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (MIT)
+//////////////////////////////////////////////////////////////////////////////
+#include "umpire/strategy/AlignedAllocator.hpp"
+
+#include "umpire/util/Macros.hpp"
+
+namespace umpire {
+namespace strategy {
+
+AlignedAllocator::AlignedAllocator(
+    const std::string& name,
+    int id,
+    Allocator allocator,
+    std::size_t alignment) :
+  AllocationStrategy(name, id),
+  m_allocator(allocator.getAllocationStrategy()),
+  m_alignment{alignment},
+  m_mask{static_cast<uintptr_t>(~(m_alignment-1))}
+{
+  if (m_allocator->getPlatform() != Platform::host) {
+    UMPIRE_ERROR("Cannot construct AlignedAllocator from non-host Allocator.");
+  }
+}
+
+void* 
+AlignedAllocator::allocate(std::size_t bytes)
+{
+  std::size_t total_bytes = bytes+sizeof(void*)+m_alignment-1;
```

Comment:
This only works for alignments that are a power of 2.  I suggest checking/throwing an error otherwise.